packages/block-library/src/social-link/edit.native.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * External dependencies
+ */
+import { View, Animated, Easing, TouchableWithoutFeedback } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { BlockControls } from '@wordpress/block-editor';
+import { useEffect, useState, useRef } from '@wordpress/element';
+import {
+	ToolbarGroup,
+	ToolbarButton,
+	LinkSettings,
+} from '@wordpress/components';
+import { usePreferredColorSchemeStyle } from '@wordpress/compose';
+import { __, sprintf } from '@wordpress/i18n';
+import { link, Icon } from '@wordpress/icons';
+/**
+ * Internal dependencies
+ */
+import { getIconBySite, getNameBySite } from './social-list';
+import styles from './editor.scss';
+
+const ANIMATION_DELAY = 300;
+const ANIMATION_DURATION = 400;
+
+const SocialLinkEdit = ( {
+	attributes,
+	setAttributes,
+	isSelected,
+	onFocus,
+} ) => {
+	const { url, service } = attributes;
+	const activeIcon =
+		styles[ `wp-social-link-${ service }` ] || styles[ `wp-social-link` ];
+
+	const inactiveIcon = usePreferredColorSchemeStyle(
+		styles.inactiveIcon,
+		styles.inactiveIconDark
+	);
+
+	const [ isLinkSheetVisible, setIsLinkSheetVisible ] = useState( false );
```

Comment:
WDYT about moving it at the top of the component? I think that it's more readable to have a declaration of state at the top than here.